### PR TITLE
Add test for result code 3324

### PR DESCRIPTION
**Pull request by emadkaramad**
Result code 3324 should raise TRPR0003

### CODE REVIEW

File: features/330-serious-violence-reduction-order-3324/pnc-data.after.xml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<CPE02>
```
Comment (bjpirt) [inline features/330-serious-violence-reduction-order-3324/pnc-data.after.xml:1]: As we haven't actually done this against the real PNC I don't think this file is valid.

File: features/330-serious-violence-reduction-order-3324/pnc-data.before.xml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<CPE02>
```
Comment (bjpirt) [inline features/330-serious-violence-reduction-order-3324/pnc-data.before.xml:1]: Also not sure we need this

File: features/330-serious-violence-reduction-order-3324/pnc-data.xml [hunk 3]
```diff
@@ -0,0 +1 @@
...
						</OffenceStart>
					</OffenceTiming>
					<ChargeDate>2010-12-02</ChargeDate>
					<ArrestDate>2010-12-01</ArrestDate>
					<LocationOfOffence>Kingston High Street</LocationOfOffence>
				</BaseOffenceDetails>
			</Offence>
		</Defendant>
	</Case>
</NewCaseMessage>
```
Comment (bjpirt) [inline features/330-serious-violence-reduction-order-3324/pnc-data.xml:1]: And this

File: features/330-serious-violence-reduction-order-3324/test.feature [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+	@Should
+	@LoadTestUI
+	@NextUI
+	Scenario: trigger is correctly generated for sexual offences (Serious Violence Reduction Order)
+		Given I am logged in as "supervisor"
+			And I view the list of exceptions
+		When I open this record
+			And I click the "Triggers" tab
+		Then I see trigger "TRPR0003" for offence "1"
+			And I see trigger "TRPR0004" for offence "1"
```
Comment (bjpirt) [inline features/330-serious-violence-reduction-order-3324/test.feature:26]: I'm not sure the test needs to test these other triggers - wonder if we should remove the things from the input data that are triggering them?

Comment (emadkaramad) [reply features/330-serious-violence-reduction-order-3324/test.feature:26]: Agreed. I'll remove them from the input message.

Comment (emadkaramad) [reply features/330-serious-violence-reduction-order-3324/test.feature]: Updated the test and the input message to only generate TRPR0003 trigger.

File: features/330-serious-violence-reduction-order-3324/pnc-data.before.xml (hunk 2 above)
Comment (emadkaramad) [reply features/330-serious-violence-reduction-order-3324/pnc-data.before.xml:1]: Updated the file with the response from PNC.

File: features/330-serious-violence-reduction-order-3324/pnc-data.after.xml (hunk 1 above)
Comment (emadkaramad) [reply features/330-serious-violence-reduction-order-3324/pnc-data.after.xml:1]: This file is also updated with the response from the PNC after returning exception. It shows that no data has changed on the PNC.

File: features/330-serious-violence-reduction-order-3324/mock-pnc-responses.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    matchRegex: "CXU02",
+    response: `<?XML VERSION="1.0" STANDALONE="YES"?>
+      <CXU02>
+        <GMH>073GENL000001RNEWREMPNCA05A73000017300000120210415154673000001                                             09000${
+          parseInt(Math.random() * 8999, 10) + 1000
+        }</GMH>
+        <TXT>I0001 - THE FOLLOWING ELEMENT(S) IN THE DIS SEGMENT CONTAIN INVALID DATA: DISPOSAL TYPE                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     </TXT>
+        <GMT>000003073GENL000001S</GMT>
+      </CXU02>`,
+    expectedRequest: "",
```
Comment (bjpirt) [inline features/330-serious-violence-reduction-order-3324/mock-pnc-responses.js:15]: If we've got this file to mock PNC requests in the emulator then we should also capture the expected request

File: features/330-serious-violence-reduction-order-3324/test.feature [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+	Background:
+		Given the data for this test is in the PNC
+			And "input-message" is received
+
+	@Should
+	@OnlyRunsOnPNC
+	Scenario: trigger is correctly generated for Serious Violence Reduction Order
+		Given I am logged in as "supervisor"
+			And I view the list of exceptions
+    Then I see exception "HO100402" for this record in the exception list
```
Comment (bjpirt) [inline features/330-serious-violence-reduction-order-3324/test.feature:22]: Indentaition is a bit off here

**Review by bjpirt — APPROVED** [review]

**Pull request merged**